monai/networks/nets/vista3d.py

```diff
@@ -0,0 +1,957 @@
+
+        """
+        image_size = input_images.shape[-3:]
+        device = input_images.device
+        if point_coords is None and class_vector is None:
+            return self.NINF_VALUE + torch.zeros([1, 1, *image_size], device=device)
+
+        bs = self.get_bs(class_vector, point_coords)
+        if patch_coords is not None:
+            # if during validation and perform enable based point-validation.
+            if labels is not None and label_set is not None:
+                # if labels is not None, sample from labels for each patch.
+                if val_point_sampler is None:
+                    val_point_sampler = self.sample_points_patch_val
+                point_coords, point_labels, prompt_class = val_point_sampler(
+                    labels, patch_coords, label_set
+                )
+                if prompt_class[0].item() == 0:
+                    point_labels[0] = -1
+                labels, prev_mask = None, None
+            elif point_coords is not None:
+                # If not performing patch-based point only validation, use user provided click points for inference.
+                # the point clicks is in original image space, convert it to current patch-coordinate space.
+                point_coords, point_labels = self.update_point_to_patch(
+                    patch_coords, point_coords, point_labels
+                )
+
+        if point_coords is not None and point_labels is not None:
+            # remove points that used for padding purposes (point_label = -1)
+            mapping_index = ((point_labels != -1).sum(1) > 0).to(torch.bool)
+            if mapping_index.any():
+                point_coords = point_coords[mapping_index]
+                point_labels = point_labels[mapping_index]
+                if prompt_class is not None:
+                    prompt_class = prompt_class[mapping_index]
+            else:
+                if self.auto_freeze or (class_vector is None and patch_coords is None):
+                    # if auto_freeze, point prompt must exist to allow loss backward
+                    # in training, class_vector and point cannot both be None due to loss.backward()
+                    mapping_index.fill_(True)
+                else:
+                    point_coords, point_labels = None, None
+
+        if point_coords is None and class_vector is None:
+            return self.NINF_VALUE + torch.zeros([bs, 1, *image_size], device=device)
+
+        if (
+            self.image_embeddings is not None
+            and kwargs.get("keep_cache", False)
+            and class_vector is None
+        ):
+            out, out_auto = self.image_embeddings, None
+        else:
+            out, out_auto = self.image_encoder(
+                input_images,
+                with_point=point_coords is not None,
+                with_label=class_vector is not None,
+            )
+        input_images = None
```

Comment:
Hi @heyufan1995 , why set input_images to None here? it will not be used in the following part

Reply:
it is not used in later part. It is set to None and release memory.